plugin/parinfer.vim

```diff
@@ -60,6 +31,19 @@ function! s:repeat(name, count)
   endif
 endfunction
 
+function! s:process(event)
+  let l:event = { "event": a:event,
+                \ "position": getpos('.'),
+                \ "lines": getline(1,line('$')),
+                \ "mode": g:parinfer_mode,
+                \ "preview_cursor_scope": g:parinfer_preview_cursor_scope }
+  let l:result = ParinferProcessEvent(l:event)
+  if l:result["lines"] !=# l:event["lines"]
```

Comment:
Is this about as fast as doing the check in cljs? I'd be surprised if you're not getting bit by serialization. Try working in https://github.com/clojure/clojure/blob/master/src/clj/clojure/core.clj